lib/babel-plugins/global-scope.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+const _path = require('path');
+/**
+ * Convert all top level variable declarations in "app.js" into explictly
+ * defined variables on the global scope.
+ */
+module.exports = function (babel) {
+	const t = babel.types;
+	let didEdits = false;
+	return {
+		name: 'app.js top level variables global transform',
+		visitor: {
+			Program: function (path, state) {
+				const logger = state.opts.logger;
+				if (_path.basename(this.file.opts.filename) !== 'app.js') {
+					return;
+				}
+				for (const bodyPath of path.get('body')) {
+					for (const name in t.getBindingIdentifiers(bodyPath.node, false, true)) {
+						didEdits = true;
+						bodyPath.insertAfter(
+							t.expressionStatement(
+								t.assignmentExpression(
+									'=',
+									t.identifier(`global.${name} `),
+									t.identifier(`${name}`)
+								)
+							)
+						);
+					}
+				}
+				if (didEdits) {
+					logger && logger.warn('The implicit global scope for variable declarations in app.js is deprecated in 7.5.0, and will be removed in 9.0.0');
```

Comment:
This warns in my 7.5.0 build although I do not use `global.XXXX` somewhere. I only use Alloy that has `Alloy.Globals` references.

Reply:
This will warn if you declare variables in the global scope and do not explicitly assign them as properties of `global`. Basically if you declare variables at the top-level of app.js , that has had the effect of silently making them global properties before. We're planning to eliminate that and make them have to be explicitly assigned.

Reply:
Is Alloy and it's `Alloy.Globals` prepared for this? I agree on the change, makes total sense.